### ISSUE TITLE
Issue with error handling with custom detectors

### ISSUE THREAD

**Issue by ericofusco** [bridgecrewio/checkov#3971]
I'm trying to run checkov from a python script with the following command:

```
subprocess.run(
  ["/app/bin/checkov", "-f", "tfplan.json", "-o", "json"], 
  capture_output=True, 
  text=True, 
  check=True, 
  encoding="utf-8")
```

I get the following error:
```
[ERROR]  Failed to get detectors from customer_run_config_response, error: 'NoneType' object is not subscriptable
```

It might be an issue when checkov is executed without environment variables, when I run manually in the container, it works fine, but whatever is happening checkov isn't handling correctly.

It looks like related to this recent pull request: https://github.com/bridgecrewio/checkov/pull/3926 cc @lirshindalman

Running checkov 2.2.112.

**Comment by ericofusco** [comment]
Actually it's not failing the scan. I just wasn't printing the results when running via script, apologies, the error is shown everytime,. I realized the error is caught and only logged.

It looks like in `customer_run_config_response['secretsPolicies']`, the field doesn't have `secretsPolicies`. I'm running without a bridgecrew API key.

**Comment by lirshindalman** [comment]
hi @ericofusco thanks for reaching out. you're right I'll look into it and add a fix.
